backend/src/channels/channels.service.ts

```diff
@@ -40,9 +52,19 @@ export class ChannelsService {
       } else {
         channel.password = undefined;
       }
+      await this.usersRepository.findOneByOrFail({
+        id: +createChannelDto.userId,
+      });
       channel.name = createChannelDto.name;
+      ret = await this.channelsRepository.save(channel);
+      role = MembershipRoleType.OWNER;
     }
-    return await this.channelsRepository.save(channel);
+    this.membershipsService.create({
+      userId: userId,
+      role: role,
+      channelId: channel.id,
+    });
+    return ret;
   }
 
   async checkDirectUsers(createChannelDto: CreateChannelDto, channel: Channel) {
```

Comment:
I would verify if the two userIds are different or both undefined: see the use of `@Check` in `friendship.entity.ts:20`